src/commands/run.rs

```diff
@@ -557,3 +591,36 @@ fn ctx_set_listenfd(num_fd: usize, builder: WasiCtxBuilder) -> Result<(usize, Wa
 
     Ok((num_fd, builder))
 }
+
+fn generate_coredump(err: &anyhow::Error, source_name: &str, coredump_path: &str) -> Result<()> {
+    let bt = err
+        .downcast_ref::<wasmtime::WasmBacktrace>()
+        .ok_or_else(|| anyhow!("failed to downcast to WasmBacktrace"))?;
+
+    let mut coredump_builder =
+        wasm_coredump_builder::CoredumpBuilder::new().executable_name(source_name);
+
+    {
+        let mut thread_builder = wasm_coredump_builder::ThreadBuilder::new().thread_name("main");
+
+        for frame in bt.frames() {
+            let coredump_frame = wasm_coredump_builder::FrameBuilder::new()
+                .funcidx(frame.func_index())
+                .build();
```

Comment:
Sorry the context is lost on the prior comment with the update here, but again while this isn't a blocker not including the function offset seems like it could hinder the usefulness of backtraces since you don't get where you are in a function, just that you're in a certain function. If you're exploring a certian `unreachable` condition, for example, there'd be no way to figure out which one in a function was hit.

Reply:
I opened https://github.com/WebAssembly/tool-conventions/pull/202 to add `funcidx` to the coredump format.

Reply:
unfortunately, I don't see a way to retrieve the instruction offset in https://docs.rs/wasmtime/6.0.0/wasmtime/struct.FrameInfo.html.

There's a  method to retrieve the func offset in the module: https://docs.rs/wasmtime/6.0.0/wasmtime/struct.FrameInfo.html#method.func_offset.

Reply:
It looks like `func_offset` is actually what is needed? The documentation reads "Returns the offset from the original wasm module’s function to this frame’s program counter." i.e., it encodes `PC - function_start`, the offset of the instruction. (The name is somewhat confusing though -- "offset in function" rather than "offset of function"!)

Reply:
Ah great, I misunderstood. Thanks

Reply:
Added `codeoffset`.